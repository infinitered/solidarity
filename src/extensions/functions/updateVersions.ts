import { SolidarityRunContext } from '../../types'
import * as pluralize from 'pluralize'

module.exports = async (context: SolidarityRunContext): Promise<void> => {
  const { map, toPairs, isEmpty, flatten } = require('ramda')
  const { solidarity, print } = context
  const {
    getSolidaritySettings,
    setSolidaritySettings,
    updateRequirement
  } = solidarity

  // load current solidarity file
  const solidaritySettings = getSolidaritySettings(context)

  // Map over requirements with option to mutate settings
  const checks = await map(
    async requirement => updateRequirement(requirement, solidaritySettings.requirements, context),
    toPairs(solidaritySettings.requirements)
  )

  // run the array of promises you just created
  await Promise.all(checks)
    .then(results => {
      const updates = flatten(results)
      if (isEmpty(updates)) {
        print.success('\n No Changes')
      } else {
        setSolidaritySettings(solidaritySettings, context)
        const ruleMessage = pluralize('Rule', updates.length, true)
        print.success(`\n ${ruleMessage} updated`)
      }
    })
    .catch(err => {
      print.error(err)
      process.exit(2)
    })

}
