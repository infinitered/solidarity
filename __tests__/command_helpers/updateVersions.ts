import updateVersions from '../../src/extensions/functions/updateVersions'

test('updateVersions exists', () => expect(updateVersions).toMatchSnapshot())
