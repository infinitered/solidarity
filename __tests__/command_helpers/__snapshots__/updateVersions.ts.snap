// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateVersions exists 1`] = `[Function]`;
